spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+                        while (derivedProvidedProducts.hasNext()) {
+                            JsonNode derivedProvidedProduct = derivedProvidedProducts.next();
+                            assertNotNull(derivedProvidedProduct.get("productId"));
+                            assertNotNull(derivedProvidedProduct.get("productName"));
+                        }
+                    }
+                    catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+
+            assertEquals(6, entitlementCount.get());
+        }
+
+        @Test
+        @DisplayName("should export entitlement certificates")
+        void shouldExportEntitlementCertificates() throws Exception {
+            int target = ENTITILEMENT_CERTIFICATES_PATH.length() - 1;
+            AtomicInteger certCount = new AtomicInteger(0);
+            export.stream()
+                .filter(entry -> entry.getName().startsWith(ENTITILEMENT_CERTIFICATES_PATH))
+                .filter(entry -> entry.getName().lastIndexOf('/') == target)
+                .forEach(entry -> {
+                    certCount.incrementAndGet();
+                    try (InputStream istream = export.getInputStream(entry)) {
+                        String cert = new String(istream.readAllBytes());
+                        assertEquals("-----BEGIN CERTIFICATE-----", cert.substring(0, 27));
+                    }
+                    catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+        }
+
+        @Test
+        @OnlyInStandalone
+        @DisplayName("should purge import records")
+        void shouldPurgeImportRecords() throws Exception {
+            try {
+                OwnerDTO importOwner = ownerApi.createOwner(Owners.random());
+                int expectedRecordCount = 11;
+                for (int i = 0; i < expectedRecordCount; i++) {
+                    // We cannot use the OpenApi generated apis for the owner import job because the
+                    // resultData in the AsyncJobStatusDTO is an object and not a string and causes
+                    // deserialization issues.
+                    Response response = Request.from(client)
+                        .setPath("/owners/{owner_key}/imports/async")
+                        .setMethod("POST")
+                        .setPathParam("owner_key", importOwner.getKey())
+                        .addQueryParam("force", "SIGNATURE_CONFLICT")
+                        .addQueryParam("force", "MANIFEST_SAME")
+                        .addHeader("Accept", "application/json")
+                        .addHeader("Content-Type", "multipart/form-data")
+                        .setFormParam(manifest.getName(), manifest)
+                        .execute();
+
+                    assertEquals(200, response.getCode());
+                    JsonNode root = ApiClient.MAPPER.readTree(response.getBody());
```

Comment:
Hrmm... I didn't expect this to be needed so soon. This could be added as part of the Response object directly as another method for fetching the body:

```
    /**
     * Attempts to process the body as a JSON-formatted UTF-8 string. If the body cannot be
     * deserialized into a JSON object, this method throws an exception.
     *
     * @throws JsonDeserializationException
     *  if the body cannot be deserialized into a JSON object
     *
     * @return
     *  a JsonNode representing the body of the response
     */
    public JsonNode deserialize() {
        try {
            return ApiClient.MAPPER.readTree(this.getBody());
        }
        catch (IOException e) { // Impl note: JsonProcessingException *is* an IOException
            throw new JsonDeserializationException(e);
        }
    }
```